source/manual/setting-up-new-rails-app.html.md

```diff
@@ -8,140 +8,161 @@ last_reviewed_on: 2018-03-22
+```
+ruby File.read(".ruby-version").strip
 
-## Bootstrapping a new application
+source "https://rubygems.org"
 
-The quickest and easiest way to start a new Rails application on the
-GOV.UK stack is to use the [govuk-rails-app-template
-repository](https://github.com/alphagov/govuk-rails-app-template).
+gem "rails", "~> 5.2"
 
-This will create a new Rails application, installing and configuring common features.
+gem "bootsnap", "~> 1"
+gem "govuk_app_config", "~> 1"
+gem "pg", "~> 1"
+gem "plek", "~> 2"
 
-From here on we'll assume you’re configuring a new
-Rails application named `myapp` that was created with the above
-script.
+group :development do
+  gem "byebug"
+  gem "govuk-lint"
+  gem "listen"
+end
 
-## Naming conventions
+group :test do
+  gem "factory_bot_rails"
+  gem "faker"
+  gem "simplecov", require: false
+end
 
-There are a few conventions for naming apps in the Puppet configuration
-and elsewhere:
+group :development, :test do
+  gem "rspec-rails"
+end
+```
 
--   Puppet classes, hieradata and CI job names should use snake\_case -
-    eg. `content_store`
--   Hostnames, deployment scripts and CI parameters should use
-    spinal-case - eg. `content-store`
+Run `bundle && rails g rspec:install` and replace `spec/*helper.rb`.
 
-## Defining the app in the development repository
+```
+## spec/rails_helper.rb
+rm spec/rails_helper.rb
 
-You first need to add the app to the [development-vm
-configuration in puppet](https://github.com/alphagov/govuk-puppet/tree/master/development-vm).
+## spec/spec_helper.rb
+require "byebug"
+require "simplecov"
 
-Add the app to the `Procfile` and assign it the next available port
-number.
+ENV["RAILS_ENV"] ||= "test"
+require File.expand_path("../../config/environment", __FILE__)
+require "rspec/rails"
 
-    myapp: govuk_setenv myapp ./run_in.sh ../myapp bundle exec rails s -p 3456
+Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
+SimpleCov.start
 
-If the app is dependent on other apps, you'll also want to add it to the
-`Pinfile` and define its dependencies.
+RSpec.configure do |config|
+  config.expose_dsl_globally = false
+  config.infer_spec_type_from_file_location!
+end
+```
 
-    process :myapp => :anotherapp
+In config, replace the content of `database.yml` with the following.
 
-To be able to access your app via a browser, you need to add it to the
-hosts list in hieradata for development. See
-[govuk-puppet](https://github.com/alphagov/govuk-puppet/blob/master/docs/adding-a-new-app.md#including-the-app-on-machines)
-for more information.
+```
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: 12
+  template: template0
 
-You should then be able to view it at <http://myapp.dev.gov.uk>.
+development:
+  <<: *default
+  database: myapp_development
+  url: <%= ENV["DATABASE_URL"]%>
 
-## Defining the app in Puppet
+test:
+  <<: *default
+  database: myapp_test
+  url: <%= ENV["DATABASE_URL"].try(:sub, /([-_]development)?$/, '_test')%>
```

Comment:
Can we switch this to be a recommended approach of TEST_DATABASE_URL. That's more of a best practice than the sub approach.